engine/src/render/deferred_shading/plugin.cpp

```diff
@@ -214,17 +270,138 @@ void cubos::engine::deferredShadingPlugin(Cubos& cubos)
                     PerScene perScene{};
                     perScene.inverseView = localToWorld.mat;
                     perScene.inverseProjection = glm::inverse(camera.projection);
+                    perScene.directionalLightWithShadowsId = -1;
 
                     perScene.ambientLight = glm::vec4(environment.ambient, 1.0F);
                     perScene.skyGradient[0] = glm::vec4(environment.skyGradient[0], 1.0F);
                     perScene.skyGradient[1] = glm::vec4(environment.skyGradient[1], 1.0F);
 
-                    for (auto [lightLocalToWorld, light] : directionalLights)
+                    int directionalLightIndex = 0;
+                    Texture2DArray directionalShadowMap = nullptr;
+                    for (auto [lightLocalToWorld, light, caster] : directionalLights)
                     {
                         auto& perLight = perScene.directionalLights[perScene.numDirectionalLights++];
                         perLight.direction = glm::normalize(lightLocalToWorld.mat * glm::vec4(0.0F, 0.0F, 1.0F, 0.0F));
                         perLight.color = glm::vec4(light.color, 1.0F);
                         perLight.intensity = light.intensity;
+
+                        if (caster.contains() && (perspectiveCamera.contains() || orthographicCamera.contains()))
+                        {
+                            float cameraZFar = 1000.0F;
+                            if (perspectiveCamera.contains())
+                            {
+                                cameraZFar = perspectiveCamera.value().zFar;
+                            }
+                            else if (orthographicCamera.contains())
+                            {
+                                cameraZFar = orthographicCamera.value().zFar;
+                            }
```

Comment:
If you do make the change I mentioned before, make sure to make this Camera type agnostic again